Source/Core/Common/MemArena.h

```diff
@@ -101,7 +101,13 @@ class MemArena final
 #ifdef _WIN32
   HANDLE hMemoryMapping;
 #else
+#ifdef ANDROID
   int fd;
+#else
+  int m_shm_fd;
+  void* m_reserved_region;
+  std::size_t m_reserved_region_size;
+#endif
 #endif
```

Comment:
```suggestion
#if defined(_WIN32)
  HANDLE hMemoryMapping;
#elif defined(ANDROID)
  int fd;
#else
  int m_shm_fd;
  void* m_reserved_region;
  std::size_t m_reserved_region_size;
#endif
```

Reply:
Additionally - `hMemoryMapping` and `fd` lack `m_` prefix, so they probably should be renamed for consistency (or avoid using prefix for new member variables too).

Reply:
Regarding the variable prefixes, I've tried to avoid touching multiple platforms in a single PR, so they'll be done one at a time. The Windows equivalent PR is #9544. There is no Android one yet because I'm not familiar with that platform.

Regarding the defines, our codebase seems to use both variants. Not sure whether it's better to check for the symbol or for a nonzero definition.

Reply:
Does new Linux approach fail to work on Android?

Reply:
Genuinely, no idea. I don't have an Android device capable of running Dolphin to test it.

Reply:
With android being Linux at its core I'd expect this approach to work, unless it is artificially restricted by permissions. Unfortunately, I don't have Dolphin-capable android device/VM/emulator either for quick testing.

Reply:
shm is not available on Android.